src/vue/common/DescriptionEditor.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <div class="description-editor">
+    <markdown-editor
+      @input="onInput"
+      :value="value"
+      :sanitize="true"
+      ref="markdownEditor"
+      :configs="configs" />
+  </div>
+</template>
+
+<script>
+import MarkdownEditor from 'vue-simplemde/src/markdown-editor'
+
+const EVENT = {
+  input: 'input',
+}
+
+export default {
+  name: 'description-editor',
+  components: { MarkdownEditor },
+  props: {
+    value: { type: [String, Number], default: '' },
+  },
+
+  data () {
+    return {
+      content: '',
+      configs: {
+        toolbar: [
+          'bold', 'italic', '|',
+          'heading-1', 'heading-2', '|',
+          'image', 'link', '|',
+          'quote', 'unordered-list', 'ordered-list', '|',
+          'preview', 'fullscreen',
+        ],
+        status: false,
+        spellChecker: false,
+      },
+    }
+  },
+
+  computed: {
+    simplemde () {
+      return this.$refs.markdownEditor.simplemde
+    },
+  },
+
+  mounted () {
+    try {
+      this.simplemde.togglePreview()
+
+      this.simplemde.codemirror.on('beforeChange', (instance, changeObj) => {
+        // do some things
+      })
+
+      this.$refs.markdownEditor.initialize() // init
+      this.simplemde.toTextArea()
+      this.simplemde.isPreviewActive() // returns boolean
+      this.simplemde.isSideBySideActive() // returns boolean
+      this.simplemde.isFullscreenActive() // returns boolean
+      this.simplemde.clearAutosavedValue() // no returned value
+      this.simplemde.markdown(this.content) // returns parsed html
+      this.simplemde.codemirror.refresh() // refresh codemirror
```

Comment:
remove excess comments, if some of them are hacks describe it implicitly. you can leave a comment with reference to the documentation to simplemde right  under the import.

Reply:
ok